sway-fmt/src/traversal_helper.rs

```diff
@@ -79,6 +79,75 @@ pub fn format_data_types(text: &str) -> String {
     })
 }
 
+/// Formats Sway data types: Enums and Structs ...but _swayfully_
+pub fn format_swayful_data_types(text: &str) -> String {
+    todo!("Use longest_var, and add spaces");
+    let longest_var: usize = find_longest_variant(text);
+    custom_format_with_comments(text, &mut move |text, result, current_char, iter| {
+        result.push(current_char);
+        match current_char {
+            '}' => {
+                clean_all_whitespace(iter);
+                if let Some((_, next_char)) = iter.peek() {
+                    if *next_char != ',' {
+                        result.push(',');
+                    }
+                }
+            }
+            ':' => {
+                let field_type = get_data_field_type(text, iter);
+                result.push_str(&field_type);
+            }
+            _ => {}
+        }
+    })
+}
+
+// Returns the length of the longest variant key name.
+fn find_longest_variant(text: &str) -> usize {
```

Comment:
you can (and should) find the longest variant/field while using `custom_format_with_comments` (maybe we rename it then to something else) since that function is already ignoring the comments by default, 

in your implementation comments are not handled at all

Reply:
Ok was wondering that, I'll work on implementing it!